pootle/static/js/editor/components/UnitSource.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) Pootle contributors.
+ *
+ * This file is a part of the Pootle project. It is distributed under the GPL3
+ * or later license. See the LICENSE file for a copy of the license and the
+ * AUTHORS file for copyright and authorship information.
+ */
+
+import React from 'react';
+
+import { t } from 'utils/i18n';
+
+
+const UnitSource = React.createClass({
+
+  propTypes: {
+    id: React.PropTypes.number.isRequired,
+    values: React.PropTypes.array.isRequired,
+    hasPlurals: React.PropTypes.bool.isRequired,
+    sourceLocaleCode: React.PropTypes.string,
+    sourceLocaleDir: React.PropTypes.string,
+  },
+
+  getUnitClass() {
+    const pluralClass = this.props.values.length > 1 ? " translate-plural" : '';
```

Comment:
Is `hasPlurals` (and `has_plural` in the backend) language-specific? I.e. is there a chance for it to be different from `this.props.values.length > 1`?

If yes, then this needs to rely on `this.props.hasPlurals`. If not, `hasPlurals` is redundant and it can be computed based on `this.props.values`.